From e35f5eeeaa4c7b9ec1ae0720fc7de0fc4d43e02f Mon Sep 17 00:00:00 2001
From: Arnaldo Garcia Rincon <agarrcia@amazon.com>
Date: Thu, 30 May 2024 14:38:33 +0000
Subject: [PATCH] oci: inject kmod in all containers

Append a new mount to the default spec created for Linux containers

Signed-off-by: Arnaldo Garcia Rincon <agarrcia@amazon.com>
---
 oci/defaults.go | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/oci/defaults.go b/oci/defaults.go
index c3dae8b..2e90cfa 100644
--- a/oci/defaults.go
+++ b/oci/defaults.go
@@ -100,6 +100,12 @@ func DefaultLinuxSpec() specs.Spec {
 				Source:      "shm",
 				Options:     []string{"nosuid", "noexec", "nodev", "mode=1777"},
 			},
+			{
+				Destination: "/sbin/modprobe",
+				Type:        "bind",
+				Source:      "/usr/bin/kmod",
+				Options:     []string{"exec", "bind", "ro"},
+			},
 		},
 		Linux: &specs.Linux{
 			MaskedPaths: []string{
-- 
2.44.0
